packages/calypso-e2e/src/lib/components/comments-component.ts

```diff
@@ -95,11 +84,14 @@ export class CommentsComponent extends BaseContainer {
 			throw new Error( `Failed to select a comment. Please check the comment number or selector.` );
 		}
 
-		await commentToLike.waitForElementState( 'stable' );
-
-		const likeButton = ( await commentToLike.$( selectors.likeButton ) ) as ElementHandle;
 		// Click the like button and wait until the animations are done.
-		await Promise.all( [ likeButton.click(), likeButton.waitForElementState( 'enabled' ) ] );
+		const likeButton = await commentToLike.waitForSelector( selectors.likeButton );
+		await Promise.all( [
+			likeButton.waitForElementState( 'stable' ),
+			// The loading attribute is added to the class temporarily while the animation is processing.
+			likeButton.waitForSelector( `${ selectors.likeButton }.loading`, { state: 'hidden' } ),
+			likeButton.click(),
+		] );
```

Comment:
I think there are 2 issues here:
1. It doesn't look like the bounding box changes for the like element (the star icon), so we should not need the stability check, and
2. The `.loading` check is potentially flaky as it can be prematurely satisfied (before the class is even added).

Let's tweak the `likeButton` selector and add the `unlikeButton` one that will help us wait for the state change:

```js
const selectors = {
	// ...
	likeButton: '.comment-not-liked > a',
	unlikeButton: '.comment-liked > a',
};
```

With that we should be able to simplify the above:
```suggestion
		const likeButton = await commentToLike.waitForSelector( selectors.likeButton );
		await likeButton.click();
		await commentToLike.waitForSelector( selectors.unlikeButton );
```

Reply:
So the reason why I structured it this way was to achieve the following:

1. use one selector to refer to the button so the code can be reused.
2. wait for the animation to complete before performing additional actions.

The suggested change does not fulfill part 1, as the sequence of calls are only good when performing the `not liked -> like` path. When attempting to go the other way with `liked -> not liked` this method will fail at this line:
`const likeButton = await commentToLike.waitForSelector( selectors.likeButton );`
because the selector referenced by `likeButton` will never be on the page for this particular comment.

For part 2, Line 90 was meant to help with waiting for the animation to complete. 
With that said however, I have taken another look at the debug logs and it appears the load state `load` is fired when the animation is completed and the `liked by` text is rendered on page. I have simplified this block of calls to a sequential call and it appears to do the trick.

I've pushed the revised block that looks like this:

```
		const likeButton = await commentToLike.waitForSelector( selectors.likeButton );
		await likeButton.click();
		await this.page.waitForLoadState( 'load' );
```